Use of signed char fails on ARM architectures

Hi,

On NVIDIA Jetson AGX (ARMv8) I do a fresh install of pygeos using Docker:

```
docker run -it --user root --rm arm64v8/ubuntu:20.04

apt-get update
apt-get install -y python3-dev python3-pip libgeos-dev
pip3 install --no-cache-dir cython 
pip3 install --no-cache-dir numpy
pip3 install --no-cache-dir --no-build-isolation pygeos
```

The following [example given in the documentation](https://pygeos.readthedocs.io/en/latest/set_operations.html) for `union_all()` method results in TypeError: 

```
import pygeos

line_1 = pygeos.Geometry("LINESTRING(0 0, 2 2)")
line_2 = pygeos.Geometry("LINESTRING(2 2, 3 3)")
pygeos.union_all([line_1, line_2])
```

Error message:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.8/dist-packages/pygeos/decorators.py", line 55, in wrapped
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pygeos/set_operations.py", line 353, in union_all
    collections = lib.create_collection(geometries, GeometryType.GEOMETRYCOLLECTION)
TypeError: One of the Geometry inputs is of incorrect geometry type.
```

Tests also fail with the same error if I install from the repository:

```
apt-get install git
pip3 install --no-cache-dir pytest
cd /tmp
git clone https://github.com/pygeos/pygeos.git
cd pygeos
pip3 install -e .[test] --no-build-isolation
pytest pygeos
```

Output:

```
=============================================== test session starts ================================================
platform linux -- Python 3.8.5, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: /tmp/pygeos
collected 18 items / 10 errors / 8 selected

====================================================== ERRORS ======================================================
________________________________ ERROR collecting pygeos/test/test_constructive.py _________________________________
pygeos/test/test_constructive.py:7: in <module>
    from .common import (
pygeos/test/common.py:22: in <module>
    geometry_collection = pygeos.geometrycollections(
pygeos/creation.py:174: in geometrycollections
    return lib.create_collection(geometries, GeometryType.GEOMETRYCOLLECTION)
E   TypeError: One of the Geometry inputs is of incorrect geometry type.
_________________________________ ERROR collecting pygeos/test/test_coordinates.py _________________________________
pygeos/test/test_coordinates.py:7: in <module>
    from .common import empty
pygeos/test/common.py:22: in <module>
    geometry_collection = pygeos.geometrycollections(
pygeos/creation.py:174: in geometrycollections
    return lib.create_collection(geometries, GeometryType.GEOMETRYCOLLECTION)
E   TypeError: One of the Geometry inputs is of incorrect geometry type.
__________________________________ ERROR collecting pygeos/test/test_creation.py ___________________________________
pygeos/test/test_creation.py:5: in <module>
    from .common import (
pygeos/test/common.py:22: in <module>
    geometry_collection = pygeos.geometrycollections(
pygeos/creation.py:174: in geometrycollections
    return lib.create_collection(geometries, GeometryType.GEOMETRYCOLLECTION)
E   TypeError: One of the Geometry inputs is of incorrect geometry type.
__________________________________ ERROR collecting pygeos/test/test_geometry.py ___________________________________
pygeos/test/test_geometry.py:6: in <module>
    from .common import point
pygeos/test/common.py:22: in <module>
    geometry_collection = pygeos.geometrycollections(
pygeos/creation.py:174: in geometrycollections
    return lib.create_collection(geometries, GeometryType.GEOMETRYCOLLECTION)
E   TypeError: One of the Geometry inputs is of incorrect geometry type.
_____________________________________ ERROR collecting pygeos/test/test_io.py ______________________________________
pygeos/test/test_io.py:8: in <module>
    from .common import all_types, point, empty_point, point_z
pygeos/test/common.py:22: in <module>
    geometry_collection = pygeos.geometrycollections(
pygeos/creation.py:174: in geometrycollections
    return lib.create_collection(geometries, GeometryType.GEOMETRYCOLLECTION)
E   TypeError: One of the Geometry inputs is of incorrect geometry type.
___________________________________ ERROR collecting pygeos/test/test_linear.py ____________________________________
pygeos/test/test_linear.py:5: in <module>
    from .common import empty_point
pygeos/test/common.py:22: in <module>
    geometry_collection = pygeos.geometrycollections(
pygeos/creation.py:174: in geometrycollections
    return lib.create_collection(geometries, GeometryType.GEOMETRYCOLLECTION)
E   TypeError: One of the Geometry inputs is of incorrect geometry type.
_________________________________ ERROR collecting pygeos/test/test_measurement.py _________________________________
pygeos/test/test_measurement.py:6: in <module>
    from .common import point_polygon_testdata
pygeos/test/common.py:22: in <module>
    geometry_collection = pygeos.geometrycollections(
pygeos/creation.py:174: in geometrycollections
    return lib.create_collection(geometries, GeometryType.GEOMETRYCOLLECTION)
E   TypeError: One of the Geometry inputs is of incorrect geometry type.
_________________________________ ERROR collecting pygeos/test/test_predicates.py __________________________________
pygeos/test/test_predicates.py:6: in <module>
    from .common import point, all_types, line_string, polygon, geometry_collection, empty
pygeos/test/common.py:22: in <module>
    geometry_collection = pygeos.geometrycollections(
pygeos/creation.py:174: in geometrycollections
    return lib.create_collection(geometries, GeometryType.GEOMETRYCOLLECTION)
E   TypeError: One of the Geometry inputs is of incorrect geometry type.
_______________________________ ERROR collecting pygeos/test/test_set_operations.py ________________________________
pygeos/test/test_set_operations.py:7: in <module>
    from .common import point, all_types, polygon, multi_polygon
pygeos/test/common.py:22: in <module>
    geometry_collection = pygeos.geometrycollections(
pygeos/creation.py:174: in geometrycollections
    return lib.create_collection(geometries, GeometryType.GEOMETRYCOLLECTION)
E   TypeError: One of the Geometry inputs is of incorrect geometry type.
___________________________________ ERROR collecting pygeos/test/test_strtree.py ___________________________________
pygeos/test/test_strtree.py:7: in <module>
    from .common import (
pygeos/test/common.py:22: in <module>
    geometry_collection = pygeos.geometrycollections(
pygeos/creation.py:174: in geometrycollections
    return lib.create_collection(geometries, GeometryType.GEOMETRYCOLLECTION)
E   TypeError: One of the Geometry inputs is of incorrect geometry type.
============================================= short test summary info ==============================================
ERROR pygeos/test/test_constructive.py - TypeError: One of the Geometry inputs is of incorrect geometry type.
ERROR pygeos/test/test_coordinates.py - TypeError: One of the Geometry inputs is of incorrect geometry type.
ERROR pygeos/test/test_creation.py - TypeError: One of the Geometry inputs is of incorrect geometry type.
ERROR pygeos/test/test_geometry.py - TypeError: One of the Geometry inputs is of incorrect geometry type.
ERROR pygeos/test/test_io.py - TypeError: One of the Geometry inputs is of incorrect geometry type.
ERROR pygeos/test/test_linear.py - TypeError: One of the Geometry inputs is of incorrect geometry type.
ERROR pygeos/test/test_measurement.py - TypeError: One of the Geometry inputs is of incorrect geometry type.
ERROR pygeos/test/test_predicates.py - TypeError: One of the Geometry inputs is of incorrect geometry type.
ERROR pygeos/test/test_set_operations.py - TypeError: One of the Geometry inputs is of incorrect geometry type.
ERROR pygeos/test/test_strtree.py - TypeError: One of the Geometry inputs is of incorrect geometry type.
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 10 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
================================================ 10 errors in 1.37s ================================================
```

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -17,6 +17,9 @@ Version 0.10 (unreleased)
 
 * Addition of a ``contains_properly`` function (#267).
 
+**Bug fixes**
+* Fixed portability issue for ARM architecture (#293)
+
 **Acknowledgments**
 
 Thanks to everyone who contributed to this release!

--- a/docker/Dockerfile.arm64
+++ b/docker/Dockerfile.arm64
@@ -1,0 +1,17 @@
+# This docker container is used for testing pygeos in ARM64 emulation mode.
+# To build it:
+# docker build . -f ./docker/Dockerfile.arm64 -t pygeos/arm64
+# Then run the pygeos test suite:
+# docker run --rm pygeos/arm64:latest python3 -m pytest -vv
+
+FROM --platform=linux/arm64/v8 arm64v8/ubuntu:20.04
+
+RUN apt-get update && apt-get install -y build-essential libgeos-dev python3-dev python3-pip --no-install-recommends
+
+RUN pip3 install numpy Cython pytest
+
+COPY . /code
+
+WORKDIR /code
+
+RUN python3 setup.py build_ext --inplace

--- a/src/ufuncs.c
+++ b/src/ufuncs.c
@@ -1988,8 +1988,7 @@ static char create_collection_dtypes[3] = {NPY_OBJECT, NPY_INT, NPY_OBJECT};
 static void create_collection_func(char** args, npy_intp* dimensions, npy_intp* steps,
                                    void* data) {
   GEOSGeometry *g, *g_copy;
-  int n_geoms, type;
-  char actual_type, expected_type, alt_expected_type;
+  int n_geoms, type, actual_type, expected_type, alt_expected_type;
 
   GEOS_INIT;
 

